packages/core/src/util.ts

```diff
@@ -51,14 +51,63 @@ export function makeBacktrace(stack: string, shift = 0): BacktraceFrame[] {
           column: line.column
         }
       })
-    backtrace.splice(0, shift)
+    if (filterHbSourceCode) {
+      backtrace.splice(0, calculateBacktraceShift(backtrace))
+    }
+
     return backtrace
   } catch (_err) {
     // TODO: log error
     return []
   }
 }
 
+const DEFAULT_BACKTRACE_SHIFT = 3
+
+function isFrameFromHbSourceCode(frame: BacktraceFrame) {
+  return frame.file.indexOf('@honeybadger-io') > -1 || frame.method.indexOf('@honeybadger-io') > -1
+}
+
+/**
+ * If {@link generateStackTrace} is used, we want to exclude frames that come from
+ * Honeybadger's source code.
+ *
+ * Logic:
+ * - For each frame, increment the shift if source code is from Honeybadger
+ * - If a frame from an <anonymous> file is encountered increment the shift ONLY if between Honeybadger source code
+ *   (i.e. previous and next frames are from Honeybadger)
+ * - Exit when frame encountered is not from Honeybadger source code
+ *
+ * Note: this will not always work, especially in browser versions where code
+ *       is minified, uglified and bundled.
+ *       For those cases we default to 3:
+ *       - generateStackTrace
+ *       - makeNotice
+ *       - notify
+ */
+export function calculateBacktraceShift(backtrace: BacktraceFrame[]) {
+  let shift = 0
+  for (let i = 0; i < backtrace.length; i++) {
+    const frame = backtrace[i]
+    if (isFrameFromHbSourceCode(frame)) {
+      shift++
+      continue
+    }
+
+    if (frame.file || frame.file === '<anonymous>') {
```

Comment:
What's the scenario where a frame's file is `<Anonymous>`? THat's missing in the explanation.

Reply:
Your comment saved me! I missed a negation (it should be `if (!frame.file ...`).

To answer your question, this came up from the fault I was debugging. I don't know when it happens, but I know it can happen. That's why I came up with the [not-so-straight-forward] approach to ignore these frames if they happen to be in-between our source code. Otherwise, the algorithm would stop here and would not see that the next frame is still honeybadger source code.

See attached image.

<img width="951" alt="image" src="https://user-images.githubusercontent.com/5813382/208142627-1284f3f1-518e-405e-82a0-15319e86cdb5.png">